### DISCUSSION TITLE
Can't activate in Visual Studio

### DISCUSSION THREAD

**Discussion by dantmnf** [community/community#13759] [Copilot Conversations]
No new page opened after this prompt.
```
---------------------------
Microsoft Visual Studio
---------------------------
GitHub Copilot

A device activation code will be copied to the clipboard. Please paste this on the GitHub website to authorize GitHub Copilot.
---------------------------
OK   Cancel   
---------------------------
```

Got this instead:
```
GitHub Copilot 1.8.0.0 (v1.8.0.0@ed17206d4)

         .'';;+++++++;;;'.            
       ';;;;;;;++++++;;;;;;'.         
     .;'.      ;++++.      .;'        
    .+;        ;;';;        '+'       
    ;++.     .'+;''+;.     .;++.      
 ..'+++;;;;;;;;.   .';;;;;;;+++;..    
'+++++;   ..           ...  '+++++;   
++++++;     .;;.    ;;'     '++++++   
++++++;     '++'   .++;     '++++++   
.';+++;      '.     .'.     '++++;'   
   .'';;;;'...        ..'';;;;'.      
       ..'';;;;;;;;;;;;;;'..          

[Error] An error occurred for 'CreateCopilotServiceAsync': System.Runtime.InteropServices.COMException (0x800401D0): OpenClipboard Failed (Exception from HRESULT: 0x800401D0 (CLIPBRD_E_CANT_OPEN))
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at System.Windows.Clipboard.Flush()
   at System.Windows.Clipboard.CriticalSetDataObject(Object data, Boolean copy)
   at System.Windows.Clipboard.SetDataInternal(String format, Object data)
   at GitHub.Copilot.CopilotServiceExtensions.<CheckSignInAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotServiceFactory.<CreateServiceAsync>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotPackage.<CreateCopilotServiceAsync>d__9.MoveNext() in /_/src/Vsix/CopilotPackage.cs:line 97
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotPackage.<<SignIn>b__10_0>d.MoveNext() in /_/src/Vsix/CopilotPackage.cs:line 108
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Shell.VsTaskLibraryHelper.<>c__DisplayClass35_0.<<FileAndForget>b__0>d.MoveNext()
```

**Comment by RodherDigital** [comment]


**Reply by pcgandhinj466** [reply]
-

**Comment by pcgandhinj466** [comment]
> Having this same error. Any solutions?

**Reply by darkstrauss** [reply]
Check out my comment below. Basically you need to start Visual Studio as Administrator, this will allow it to open up the window in a browser for filling in the code that gets copied to your clipboard

**Comment by darkstrauss** [comment]
Same problem here. I'm trying to use Copilot in a Unity project that I had already initialized. When I click to sign in though visual studio using the Copilot plugin I get the exact same error message.

**Reply by darkstrauss** [reply]
I resolved the issue. Start Visual Studio as an Administrator. Then open your desired project, and then press sign in. I still had the issue that it didn't want to copy the code to clipboard, but it did finally open the login page where you fill in the code to activate the plugin!

**Reply by RodherDigital** [reply]
U're lucky. I wasn't able to open the login page even launching as an admin. I was able to do it on JetBrains however, but not in VS2022 :/

**Reply by JMC2002** [reply]
I opened Visual Studio with administrator permission as you said and finally succeeded!! I've been struggling with this problem for two months!

**Comment by pipic1** [comment]
Same thing for me with Visual Studio 2022. Start VS as Administrator, then this window is prompted:
![image](https://user-images.githubusercontent.com/15725317/161007976-9230ae71-0e27-4904-ba5e-c9d6a694e879.png)
When I press OK, nothing happened, and clipboard is empty...

For other person that can access to the "device activation code", but not the web page,I think you can enter it here:
[Device Activation](https://github.com/login/device)

**Reply by henrycates** [reply]
I wish they would just give you the code in the dialog instead of needing a workaround.

**Comment by kzu** [comment]
The latest version will properly detect this issue and not fail. It will show the activation code in the dialog, output window and status bar, so you can manually navigate to https://github.com/login/device and enter the code there.

**Reply by RobotX5** [reply]
I have version 1.14.0.1 and it does not have the activation code in the output. did that change.
or is there a new verison

**Reply by ildarcpsi** [reply]
Thanks. Only your answer helped.

**Comment by tolgabalci** [comment]
Installed latest version.  After restart and loading of project, I see the dialog (no activation code is shown on the dialog). Pressing ok does not open any other page/browser and nothing is copied in to the clipboard.

![image](https://user-images.githubusercontent.com/2467618/165218526-08cebad4-343d-4d31-83b8-3af24ea49630.png)

I tried full laptop restart and admin mode.  Made no difference.

Here is the GitHub Copilot output:
GitHub Copilot 1.12.0.1 (v1.12.0.1@c51c72a6a)
GitHub Copilot Agent v1.12.0
NotSignedIn

I'm using Windows 10 with Visual Studio Enterprise 17.1.5

Having to give up using it for now.  There is no way to get it working that I can see.

**Reply by RobotX5** [reply]
following - I have this same issue.

**Reply by mfcallahan** [reply]
This has been my experience for the last few months as well. I have yet to find a resolution and I gave up about 6 weeks ago. GitHub CoPilot works really well in VS Code, and I figured I'd give it another shot in Visual Studio, but still no dice. Can't get around this copy/paste authorization thing...

**Reply by NoCodeMaster** [reply]
following - I have this same issue.

Any news?

**Reply by hg98-park** [reply]
same problem here.
Looks like it has something to do with corporate network.
I found this message in the long list of error message 
"self signed certificate in certificate chain"

**Reply by ghost** [reply]
Crazy that they have yet to fix this... I have the same issue and yeah it is the corporate network smh

**Comment by VanBuren77** [comment]
I am having the same issue, I started as admin, but the code will not copy

**Reply by henrycates** [reply]
Same

**Comment by gkl-li** [comment]
Solved: Login to your corresponding microsoft account in VS, close the editor and reopen, try to get this prompt window to show again, and now look at the console for the exact code

**Comment by ricnader** [comment]
Tried every thing was said here. Nothing.

**Reply by henrycates** [reply]
Same

**Reply by ayim** [reply]
@ricnader @henrycates sounds frustrating. got a moment to screenshare? I'd love to investigate.

email me at ayim at github dot com and we'll get the ball rolling.

**Comment by dwmcmurrian** [comment]
I had to disconnect from my VPN to get the code and window to pop up.

**Comment by Aaron-Day** [comment]
I appreciate the simplicity of the GitHub Copilot activation process, but I believe it could be improved to accommodate a wider range of scenarios.

In my situation, the activation process required me to be logged in as an administrator. Unfortunately, my administrator account does not have a license for the version of Visual Studio I am using, which prevents me from logging in to obtain the device activation code. Additionally, I have noticed that other users have encountered issues while connected to a VPN. Since my work computer requires VPN access, I am concerned that I may face this problem as well.

To address these issues, I suggest including the activation code and URL directly within the activation dialog. This would enable users facing similar challenges to complete the activation process and provide a workaround for those who cannot use the current activation method.

**Comment by arkadiuszmakarenko** [comment]
I have exactly the same issues, and it looks that it has been ongoing thing for a while.

**Comment by ghost** [comment]
If you are using a company laptop that has some corporate AD then it probably won't work and give you the "Request signInInitiate failed with message: self signed certificate in certificate chain". Welcome to Microsoft! Please join our Microsoft event to see this piece of work in action. Then sign up for that yearly enterprise subscription. Then please use our state of the art IDE where you can install this piece of art. Oh? It doesn't work? Sorry! Go use VS Code. We're working on Gopilot X://

**Comment by manm15** [comment]
#Got this issue too...

[DEBUG] [agent] [2023-05-23T09:52:09.136Z] Agent service starting
[DEBUG] [agent] [2023-05-23T09:52:10.356Z] Telemetry initialized
[Error] An error occurred for 'RunOperation:SignInInitiateAsync': StreamJsonRpc.RemoteInvocationException: Request signInInitiate failed with message: self signed certificate in certificate chain
   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__151`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotClient.<RunOperation>d__20`1.MoveNext()
[DEBUG] [agent] [2023-05-23T09:52:48.962Z] Agent service starting
[DEBUG] [agent] [2023-05-23T09:52:50.016Z] Telemetry initialized
[Error] An error occurred for 'RunOperation:RunOperation': StreamJsonRpc.RemoteInvocationException: Request signInInitiate failed with message: self signed certificate in certificate chain
   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__151`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotClient.<RunOperation>d__20`1.MoveNext()

**Comment by Novodes** [comment]
For me, there was no need to copy anything. Perhaps the process changed as of September 2023. But I also had this problem and the issue was simply going to your user profile at the top right of the window in visual studio and then you will see a GitHub icon:
![image](https://github.com/community/community/assets/35224049/4e70e6a8-2451-4ebb-b7d6-25f42fc984fa)

I had next to the GitHub icon a link to re authenticate with GitHub. I clicked it, a link opened in my browser and I got authenticated and all the problems went away.
